packages/api/src/command/medical/patient/__tests__/consolidated-get.test.ts

```diff
@@ -53,3 +57,103 @@ describe("getConsolidatedPatientData", () => {
     expect(fhir_searchResourcePages).toHaveBeenCalledTimes(1);
   });
 });
+
+describe("getCurrentConsolidatedProgress", () => {
+  it("return undefined when there is no consolidated progress", () => {
+    const resp = getCurrentConsolidatedProgress(undefined, {});
+
+    expect(resp).toBeUndefined();
+  });
+
+  it("return undefined when consolidated progress is present but params dont match", () => {
+    const consolidatedQueryProgress = makeConsolidatedQueryProgress();
+    const resp = getCurrentConsolidatedProgress([consolidatedQueryProgress], {});
+
+    expect(resp).toBeUndefined();
+  });
+
+  it("return undefined when consolidated progress is present and params match but is completed", () => {
+    const consolidatedQueryProgress = makeConsolidatedQueryProgress({
+      status: "completed",
+    });
+    const resp = getCurrentConsolidatedProgress([consolidatedQueryProgress], {
+      resources: consolidatedQueryProgress.resources,
+      dateFrom: consolidatedQueryProgress.dateFrom,
+      dateTo: consolidatedQueryProgress.dateTo,
+      conversionType: consolidatedQueryProgress.conversionType,
+    });
+
+    expect(resp).toBeUndefined();
+  });
+
+  it("return consolidated progress when consolidated progress is present and params match", () => {
+    const consolidatedProgress = makeConsolidatedQueryProgress();
+
+    const resp = getCurrentConsolidatedProgress([consolidatedProgress], {
+      resources: consolidatedProgress.resources,
+      dateFrom: consolidatedProgress.dateFrom,
+      dateTo: consolidatedProgress.dateTo,
+      conversionType: consolidatedProgress.conversionType,
+    });
+
+    expect(resp).toEqual(consolidatedProgress);
+  });
+
+  it("return consolidated progress when consolidated progress is present with empty resource array and resources is the full list", () => {
+    const consolidatedProgress = makeConsolidatedQueryProgress({
+      resources: [...resourcesSearchableByPatient],
+    });
+
+    const resp = getCurrentConsolidatedProgress([consolidatedProgress], {
+      resources: [],
+      dateFrom: consolidatedProgress.dateFrom,
+      dateTo: consolidatedProgress.dateTo,
+      conversionType: consolidatedProgress.conversionType,
+    });
+
+    expect(resp).toEqual(consolidatedProgress);
+  });
+
+  it("return consolidated progress when consolidated progress is present with full resource array and resources is empty", () => {
+    const consolidatedProgress = makeConsolidatedQueryProgress();
```

Comment:
Is it fine to create a default at the top of the test in the beforeEach? And use that accordingly?

Reply:
Its totally fine if we want a constant input. But some tests need a different input, so in those cases I think we should create it on the function itself.

WRT to using `beforeEach`:
- for regular stuff like creating a simple object, we can just declare the `const` at the start of the file, no need to use `beforeXxxx()`;
- `beforeEach` is useful when we want to run some code before each test - which means it might run 10s of times per file;
- that last point means we should try to use `beforeAll` instead of `beforeEach` when possible, so we minimize the cost of running tests - leading to faster tests, which helps with DX and CICD.